### PR TITLE
Issue #85 - Update Angular to 2.0.1. 

### PR DESCRIPTION
**Pull request by joshuawilson**
This update also updated TypeScript to 2.0.3.

To try this change you will need to:

rm -rf node_modules
npm cache clean
npm install

### CODE REVIEW

File: src/app/board/board.component.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@ import { WorkItemService } from '../work-item/work-item.service';
 
 @Component({
   selector: 'my-board',
-  templateUrl: '/board.component.html',
-  styleUrls: ['/board.component.scss']
+  templateUrl: './board.component.html',
+  styleUrls: ['./board.component.scss']
```
Comment (sanbornsen) [inline src/app/board/board.component.ts]: Are these failing for the missing '.' after update? I bet './' is the standard way.

Comment (joshuawilson) [inline src/app/board/board.component.ts]: They do not appear to fail but you are right, the "./" is the standard way and so I updated it to match that standard.

Comment (sanbornsen) [inline src/app/board/board.component.ts]: Okay.

**Comment by nimishamukherjee** [comment]
@joshuawilson , followed the steps mentioned in the PR.
Observations:
* The UI works as expected
* Lots of errors are displayed for me
clierrors

**Comment by joshuawilson** [comment]
Try deleting the typings.json file. It has been removed in the PR but it will not delete it locally.

**Comment by nimishamukherjee** [comment]
@joshuawilson the typings.json file is not there. I did a fresh install in a new folder and tried and got the same results.

File: package.json [hunk 2]
```diff
@@ -26,45 +26,44 @@
...
-    "postinstall": "typings install",
     "prestart": "npm run clean-and-copy",
     "prebuild": "npm run clean-and-copy",
     "prebuild:prod": "npm run clean-and-copy",
+    "protractor": "sleep 15 && ./node_modules/protractor/bin/protractor protractor.config.js",
     "setup": "npm install && npm run validate",
     "start": "webpack-dev-server --inline --progress --host 0.0.0.0 --port 8088",
     "test": "karma start",
-    "typings": "typings",
+    "testserver-start": "webpack-dev-server --inline --progress --host 0.0.0.0 --port 8088",
```
Comment (ppitonak) [inline package.json]: There is inconsistency in naming npm scripts, e.g. watch:test and docker:run vs. clean-dist and testserver-start

File: package.json [hunk 3]
```diff
@@ -26,45 +26,44 @@
...
-    "protractor": "sleep 15 && ./node_modules/protractor/bin/protractor protractor.config.js",
-    "functests": "npm-run-all -p webdriver-start testserver-start protractor"
+    "watch:test": "npm test -- --auto-watch --no-single-run"
   },
   "dependencies": {
-    "@angular/common": "2.0.0-rc.6",
-    "@angular/compiler": "2.0.0-rc.6",
+    "@angular/common": "2.0.1",
+    "@angular/compiler": "2.0.1",
     "@angular/compiler-cli": "0.6.0",
```
Comment (ppitonak) [inline package.json]: Why not to update this dependency? Latest version is 0.6.3

**Comment by ppitonak** [comment]
@joshuawilson @nimishamukherjee there are similar warning in Jenkins log https://ci.centos.org/job/devtools-almighty-ui/109/consoleFull

**Comment by nimishamukherjee** [comment]
@joshuawilson the errors do not appear now locally. 
Note: typings folder needs to be deleted.

File: package.json (hunk 3 above)
Comment (joshuawilson) [inline package.json]: I updated it, thanks.

File: package.json (hunk 2 above)
Comment (joshuawilson) [inline package.json]: Changing these names should be done in a different PR.

**Comment by joshuawilson** [comment]
@ppitonak I have pushed several updates, are you still seeing the issue?

**Comment by ppitonak** [comment]
Looking in log [1], there are no such warning anymore.

[1] https://ci.centos.org/job/devtools-almighty-ui/120/console
